include/IECore/Platform.h

```diff
@@ -0,0 +1,44 @@
+//////////////////////////////////////////////////////////////////////////
+//
+//  Copyright (c) 2018, Image Engine Design Inc. All rights reserved.
+//
+//  Redistribution and use in source and binary forms, with or without
+//  modification, are permitted provided that the following conditions are
+//  met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+//     * Neither the name of Image Engine Design nor the names of any
+//       other contributors to this software may be used to endorse or
+//       promote products derived from this software without specific prior
+//       written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifdef _WIN32
+	// Windows defines SearchPath
+	#ifdef SearchPath
+		#undef SearchPath
+	#endif
+	// Windows has a different environment separator
+	#define IECORE_ENVSEP ";"
```

Comment:
Ugh. Is this only used in the SearchPath constructors? If so, I'm inclined to say that we should rejig the SearchPath class a little so that it doesn't expose separators as arguments anywhere - instead it can just automatically choose the right separator internally. I can look into this if you like...

Reply:
I'm cool with doing this, it seems easy enough but will this limit the use of it in other places vs just adding a constructor which doesn't use a separator and will choose ":" or ";" based on a _WIN32 ifdef to setPaths?

Reply:
For all the uses I'm aware of, the separator argument is useless - the documentation for it would have to be "here you must pass the right separator for the current platform". In other words, it's just a trap to fall into. I'd rather just remove the trap - if more esoteric use cases come up in the future, it's still possible to manipulate the `.paths` member directly anyway. I'll take care of this as a separate PR...